tests/cases/stream_test.py

```diff
@@ -49,7 +49,7 @@ def __init__(self):
     @access.public
     def inputStream(self, params):
         # Read body 8 bytes at a time so we can test chunking a small body
-        for chunk in iterBody(8):
```

Comment:
This is the main substantive point I want to discuss about this feature. The underlying cherrypy API gives us two options for reading these chunked TE requests, and I'm not sure which one makes for a better API, or if we really need to expose both. Hoping folks can weigh in.

1. Block until the last byte of the current chunk is read. This allows clients to (perhaps) intellegently decide what should be encoded in each chunk, which could be useful if we expect significant delays between discrete documents being sent over the socket. The drawback here is that the server side logic no longer has any control over the read buffer size, and also we have to make use of a private method in cherrypy, although I'm not that worried about using the private method since our testing is sufficient.
2. Block on read until a certain number of bytes (decided by the server logic) is available. The drawback here is that we will block until that number of bytes is available, which may be a larger buffer than the chunk size being sent by the client, meaning we might be sitting there waiting while we actually have discrete documents we could be processing.

If we choose to support both options, we just have to have a very long docstring describing all of these behaviors. I want someone to be able to call the `iterBody` method regardless of whether the client sent the body using TE: chunked or a normal known-length content body.

I guess a third option is to try and dig further down to see if we could add a timeout to the blocking read, but from my browsing cherrypy internals for a while, it didn't look simple.